USAGE.md

```diff
@@ -125,76 +145,132 @@ For the `aws-vault login` subcommand:
-those one-off temporary credentials with limited access to connect to S3 and drop some files there.
-
-In such a situation, if you are running a local server, e.g. for dev, and want to call this API,
-then you can't use an AWS session, because AWS will return a 403 on the GetFederationToken
-operation. That is when you'll use the less secure solution described above.
+## Assuming a role for more than 1h
 
-
-## Rotating Credentials
+If you try to assume a role from an opened (temporary) session, AWS considers that as *role
+chaining* and it limits your ability to assume the target role to only **1h**. Trying to use
+`--assume-role-ttl` with a value bigger than **1h** will result in an error:
+```
+aws-vault: error: Failed to get credentials for default (source profile for pix4d): ValidationError:
+The requested DurationSeconds exceeds the 1 hour session limit for roles assumed by role chaining.
+        status code: 400, request id: aa58fa50-4a5e-11e9-9566-293ea5c350ee
+```
+There are reasons though where you'd like to assume a role for a longer period. For example, when
+using a tool like [Terraform](https://www.terraform.io/), you need to have AWS credentials available
+to the application for the entire duration of the infrastructure change. And in large setups, or for
+complex resources, this can take more than 1h.  
+There are 2 solutions:
+
+1. Call aws-vault with `--no-session`. This means that the `AssumeRole` API
+will be called by using directly the IAM user credentials and not opening a session. This is not a
+*role chaining* and therefore you can request a role for up to 12 hours (`--assume-role-ttl=12h`),
+so long as you have setup your role to allow such a thing (AWS role are created by *default* with a
+max TTL of 1h). The drawback of this method is related to **MFA**. Since you are not using the AWS
+session, which is cached by `aws-vault`, if you use **MFA** (and you should), you'll have to enter
+your **MFA** token at every invocation of the `aws-vault` command. This can become a bit tedious.  
+
+2. Start `aws-vault` as a server (`aws-vault exec <profile> -s`). This will start a background
+   process that will immitate the [metadata
+endpoint](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html) that you
+would have on an EC2 instance. When your application will want to connect to AWS and fail to find
+credentials (typically in env variables), it will instead contact this server that will issue a new
+set of temporary credentials (using the same profile as the one the server was started with). This
+server will work only for the duration of the session (`--session-ttl`).  
+Note that this approach has the **major drawback** that while this `aws-vault` server runs, any
+application wanting to **connect** to AWS will be able to do so **implicitely**, with the profile the
+server was started with. Thanks to `aws-vault`, the credentials are not exposed, but the ability to
+use them to connect to AWS is!
```

Comment:
I'll let you confirm this. I think indeed aws-vault run in server mode will work for the entire duration of the session, thus solving the issue but I'm not 100% sure.
I have tried though to use it after it had been up for more than 1h and it seemed to work, so I assume this is ok

Reply:
Yes this is correct AFAIK